src/Microsoft.AspNetCore.SignalR.Core/HubEndPoint.cs

```diff
@@ -32,16 +34,32 @@ public class HubEndPoint<THub> where THub : Hub
             _lifetimeManager = lifetimeManager;
             _loggerFactory = loggerFactory;
             _hubOptions = hubOptions.Value;
+            _globalHubOptions = globalHubOptions.Value;
             _logger = loggerFactory.CreateLogger<HubEndPoint<THub>>();
             _userIdProvider = userIdProvider;
             _dispatcher = dispatcher;
         }
 
         public async Task OnConnectedAsync(ConnectionContext connection)
         {
-            var connectionContext = new HubConnectionContext(connection, _hubOptions.KeepAliveInterval, _loggerFactory);
+            // We check to see if HubOptions<THub> are set because those take precedence over global hub options.
+            // Then set the keepAlive and negotiateTimeout values to the defaults in HubOptionsSetup incase they were explicitly set to null.
+            var keepAlive = _hubOptions.KeepAliveInterval ?? _globalHubOptions.KeepAliveInterval ?? HubOptionsSetup.DefaultKeepAliveInterval;
+            var negotiateTimeout = _hubOptions.NegotiateTimeout ?? _globalHubOptions.NegotiateTimeout ?? HubOptionsSetup.DefaultNegotiateTimeout;
+            var supportedProtocols = _hubOptions.SupportedProtocols;
+            if (supportedProtocols == null && _globalHubOptions.SupportedProtocols != null)
```

Comment:
You could replace this if statement with a null coalesce on the line above. It doesn't matter if `_globalHubOptions.SupportedProtocols` protocols is null based on the check line 55.